pkg/collector/tcp.go

```diff
@@ -49,24 +49,20 @@ func (cp *CollectingProcess) startTCPServer() {
 					return
 				}
 			}
-			go cp.handleTCPClient(conn, stopCh)
+			go cp.handleTCPClient(conn)
```

Comment:
>When closing connections, we are not supposed to depend on cp.stopChan because it will cause firstly exit from startTCPServer() before we are able to take care of closing connection in handleTCPClient().

What is the reason for not closing the connection properly? We do have to defer call for connection close in L62 in `handleTCPClient` method.
Is it because the connection is created in other go routine?

Reply:
I think so. when we close `stopChan`, the signal in `stopChan` is not caught by `handleTCPClient` so `handleTCPClient` is not able to return correctly before this function `startTCPServer()` exits. Also detect the same condition at Antrea side.

Reply:
Maybe this is because of the reason that we exit the parent go routine (inline one that calls handleTCPClient) before all the other client go routines exit. Could we use wg.Add and wg.Wait here to resolve that. 
I think that may work. What do you think?

Reply:
Yes, using wait group to ensure connections are safely closed before exiting makes sense.